plugins/UsersManager/API.php

```diff
@@ -264,6 +267,64 @@ private function getDefaultUserPreference($preferenceName, $login)
         }
     }
 
+    /**
+     * Returns all users with their role for $idSite.
+     *
+     * @param int $idSite
+     * @param int|null $limit
+     * @param int|null $offset
+     * @param string|null $filter_search text to search for in the user's login, email and alias (if any)
+     * @param string|null $filter_access only select users with this access to $idSite. can be 'noaccess', 'some', 'view', 'admin', 'superuser'
+     *                                   Filtering by 'superuser' is only allowed for other superusers.
+     * @return array
+     */
+    public function getUsersPlusRole($idSite, $limit = null, $offset = 0, $filter_search = null, $filter_access = null)
```

Comment:
random thought: wouldn't it be possible to call in the API `UsersManager.getUsers` and then either one bulk API request for each user to get the related role/and capability (something like `getSitesAccessFromUser($login)`) or to call one API `getSitesAccessFromUsers($logins)`. Also it returns roles plus capabilities so maybe wonder if it should better return `getUsersPlusAccess`? 

The `Plus` in the method suggests though that it should be two different methods.

Reply:
`$filter_access` filters based on the user's access to `$idSite`, so it needs to get the user + the role for `$idSite`. Not sure what to rename it to.